thetis/shallowwater_eq.py

```diff
@@ -708,6 +721,18 @@ def residual(self, uv, eta, uv_old, eta_old, fields, fields_old, bnd_conditions=
         return -f
 
 
+class BathymetryDisplacementMassTerm(ShallowWaterContinuityTerm):
+    """
+    Contribution to water level mass term of bathymetry displacement function, used in wetting and drying.
+    Note, that this is a mass term that only takes eta as argument (unlike all other shallow water terms)
```

Comment:
You should probably change the interface to be compliant with the rest of the terms. This function is only used here so it's easy to change.